### ISSUE TITLE
Version 0.5.x should document changes in default pickle protocol

### ISSUE THREAD

**Issue by rcsalvador** [honzajavorek/redis-collections#112]
When using it with the ElasticCache Redis, the `0.5.0` is no longer working.
I fixed my application with the `0.4.2` to keep all working.
A simple `Dict` instance didn't returns any value, using square brackets or a `.get()`.
Although when traversing the dictionary using a for, the key is returned and also a `len()` returns the correct size of the dictionary.

**Comment by bbayles** [comment]
What version of the [Python `redis` package](https://github.com/andymccurdy/redis-py) are you using?

**Comment by rcsalvador** [comment]
I'm using the latest from the PyPi, it's the 3.0.1

**Comment by bbayles** [comment]
OK, thanks.

Can you provide any other info on the versions? Python version? Redis server version? Any other AWS settings would be helpful.

Also, if you could show an example of the `for` loop showing a key that you can't access with `[]` that would be useful. And what does `redis` show the key as?

One more question: are you attaching to a dict that was created previously, with another version?

**Comment by rcsalvador** [comment]
I'm using Python 3.6.6, the ElasticCache Redis is the 4.0.10 with 3 nodes, but I it just publish in only one endpoint. When I try the code:

```python
redis_client = StrictRedis(host=REDIS_HOST, port=REDIS_PORT, db=0)
my_dict = Dict(redis=redis_client, key='my_huge_dict')

# This loop works and returns the key and the value correctly
for key, value in my_dict.items():
   assert key is not None
   assert value is not None
   print(key)

assert len(my_dict) > 0        # This also works fine and returns the right value

# Then using a key value from the previous...
assert '12345' in my_dict      # This always returns false...
value = my_dict['12345']       # ...and with it, I get a KeyError
```

The same code is executed with the `0.4.2`, and it works fine.

> One more question: are you attaching to a dict that was created previously, with another version?

This question worries me, once I loaded my redis instance with more than 2 million values using the version `0.4.2` and with it, it's working well. I fixed my project to use this version to keep it working.
Do you mean that the dictionary content isn't compatible between the `0.4.2` and `0.5.0`?

**Comment by bbayles** [comment]
Thanks for the info.

> Do you mean that the dictionary content isn't compatible between the 0.4.2 and 0.5.0?

I don't think it's not compatible; I just want to try to understand the issue. Changing Python versions could cause incompatibility, but not `redis-collections` versions.

No `Dict `code changed for 0.5.0; I expect this is down to a difference in `redis-py`. Check the PRs; nothing there has changed for quite a while.

The keys are Python `pickle` objects, which get un-`pickle`d on the way back from Redis. So if a lookup is failing, there's probably some issue in how the new `redis-py` handles sending the key.

---

Do you have an Elasticache cluster that I can connect to via a jump box or something? I would like to try to solve this problem, and seeing it in action would let me do it and this would be the fastest way.

**Comment by bbayles** [comment]
OK, I figured it out.

In https://github.com/honzajavorek/redis-collections/pull/101 I made the collections use the highest pickle protocol by default.

Do this to access your previously-created dict with the new version:

```
my_dict = Dict(redis=redis_client, key='my_huge_dict', protocol=None)
```

That should work.

---

Thanks for bringing this to my attention. I should have documented this change in the release notes. I'll make an update shortly to do that.
